### PR TITLE
[query] Configurable metric metadata stats count

### PR DESCRIPTION
**Pull request by esmet**
Allows metric metadata stats to be disabled via config or request header.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/query/storage/m3/storage.go [hunk 1]
```diff
@@ -126,6 +127,33 @@ func (s *m3storage) Name() string {
...
+		nameTag := []byte{}
+		for _, label := range series.Labels {
+			// Check for both the rollup tag and the metric name label.
+			if bytes.Equal(label.Name, rollupTagName) {
+				if bytes.Equal(label.Value, rollupTagValue) {
+					rollup = true
+				}
+			} else if bytes.Equal(label.Name, promDefaultName) {
+				nameTag = label.Value
+			}
```
Comment (arnikola) [inline src/query/storage/m3/storage.go:146]: May be worth shortcircuting here if we've found both; or can use the fact that these are sorted to shortcircuit more generally

**Review by arnikola — APPROVED** [review]
LGTM!

File: src/query/storage/m3/storage.go (hunk 1 above)
Comment (esmet) [reply src/query/storage/m3/storage.go:146]: This is a good suggestion and I'm going to think about creating a pattern for looking up "internal" tags (those that start with __), possibly in bulk through some select mechanism. For now I'm going to keep this unoptimized though.

Once I come up with a decent proposed pattern I'll submit a PR and migrate other use cases.

Comment (esmet) [reply src/query/storage/m3/storage.go:146]: This is a good suggestion and something I will look into as a followup. I think we should have some sort of lookup function for internal tags that knows to stop iterating after the end of the "__"-prefixed tags. That's simple enough and gives us most of the win, I think.

File: src/query/block/meta.go [hunk 2]
```diff
@@ -139,25 +140,51 @@ type ResultMetadata struct {
...
+	}
+	m.namespaces[namespace] = struct{}{}
+}
+
+// GetNamespaces returns an array representing the set of namespaces added via AddNamespace.
+func (m ResultMetadata) GetNamespaces() []string {
+	if m.namespaces == nil {
+		return []string{}
+	}
+	namespaces := []string{}
```
Comment (rallen090) [inline src/query/block/meta.go:160]: nit: could alloc to the len of namespaces

**Review by rallen090 — APPROVED** [review]

File: src/query/block/meta.go (hunk 2 above)
Comment (esmet) [reply src/query/block/meta.go:160]: I'll follow up with an optimization PR that contains this and the internal tag lookup search short-circuit.

**Pull request merged**